ts/src/renderer/phaser/classes/PhaserUnit.ts

```diff
@@ -1,250 +1,189 @@
-class PhaserUnit extends Phaser.GameObjects.Container {
+class PhaserUnit extends PhaserAnimatedEntity {
 
+	scene: Phaser.Scene;
 	sprite: Phaser.GameObjects.Sprite;
 	label: Phaser.GameObjects.Text;
 	chat: PhaserChatBubble;
 
+	gameObject: Phaser.GameObjects.Container;
 	attributes: PhaserAttributeBar[] = [];
 
-	private followListener: EvtListener;
-	private stopFollowListener: EvtListener;
 
-	private playAnimationListener: EvtListener;
-
-	private updateLabelListener: EvtListener;
-	private hideLabelListener: EvtListener;
-
-	private renderAttributesListener: EvtListener;
-	private updateAttributeListener: EvtListener;
-
-	private renderChatListener: EvtListener;
 
 	constructor (scene: Phaser.Scene,
-				 private unit: Unit) {
+				 entity: Unit) {
 
-		super(scene);
+		super(scene, entity, `unit/${entity._stats.type}`);
 
-		const key = `unit/${unit._stats.type}`;
-
-		const sprite = this.sprite = scene.add.sprite(0, 0, key);
-		this.add(sprite);
+		this.scene = scene;
+		const translate = entity._translate;
+		this.gameObject = scene.add.container(
+			translate.x,
+			translate.y,
+			[ this.sprite ]
+		);
 
 		const label = this.label = scene.add.text(0, 0, 'cccccc');
 		label.setOrigin(0.5);
-		this.add(label);
-
-		scene.add.existing(this);
-
-		this.followListener = unit.on('follow', () => {
-			console.log('PhaserUnit follow', unit.id()); // TODO remove
-			const camera = scene.cameras.main as Phaser.Cameras.Scene2D.Camera & {
-				_follow: Phaser.GameObjects.GameObject
-			};
-			if (camera._follow === this) {
-				return;
-			}
-			camera.startFollow(this, true, 0.05, 0.05);
+		this.gameObject.add(label);
+
+		//const attributes = this.attributes;
+
+		Object.assign(this.evtListeners, {
+			followListener: entity.on('follow', this.followListener, this),
```

Comment:
keys need to be event names, not method names, because they are used for removing listeners properly.